docs/remote.md

```diff
@@ -1,18 +1,191 @@
+
+    {
+        "Err": string
+    }
+
+The string value supplied may appear in logs, so should not include confidential information.
+
+### Handshake
+
+When loaded, a remote driver process receives an HTTP POST on the URL `/Plugin.Activate` with no payload. It must respond with a manifest of the form
+
+    {
+        "Implements": ["NetworkDriver"]
+    }
+
+Other entries in the list value are allowed; `"NetworkDriver"` indicates that the plugin should be registered with LibNetwork as a driver.
+
+### Create network
+
+When the proxy is asked to create a network, the remote process shall receive a POST to the URL `/NetworkDriver.CreateNetwork` of the form
+
+    {
+        "NetworkID": string,
+        "Options": {
+            ...
+        }
+    }
+
+The `NetworkID` value is generated by LibNetwork. The `Options` value is the arbitrary map given to the proxy by LibNetwork.
+
+The response indicating success is empty:
+
+    `{}`
+
+### Delete network
+
+When a network owned by the remote driver is deleted, the remote process shall receive a POST to the URL `/NetworkDriver.DeleteNetwork` of the form
+
+    {
+        "NetworkID": string
+    }
+
+The success response is empty:
+
+    {}
+
+### Create endpoint
+
+When the proxy is asked to create an endpoint, the remote process shall receive a POST to the URL `/NetworkDriver.CreateEndpoint` of the form
+
+    {
+        "NetworkID": string,
+        "EndpointID": string,
+        "Options": {
+            ...
+        },
+        "Interfaces": [{
+            "ID": int,
+            "Address": string,
+            "AddressIPv6": string,
+            "MacAddress": string
+        }, ...]
+    }
+
+The `NetworkID` is the generated identifier for the network to which the endpoint belongs; the `EndpointID` is a generated identifier for the endpoint.
+
+`Options` is an arbitrary map as supplied to the proxy.
+
+The `Interfaces` value is a list with values of the form given. The fields in the `Interfaces` entries may be empty; and the `Interfaces` list itself may be empty. If supplied, `Address` is an IPv4 address and subnet in CIDR notation; e.g., `"192.168.34.12/16"`. If supplied, `AddressIPv6` is an IPv6 address and subnet in CIDR notation. `MacAddress` is a MAC address as a string; e.g., `"6e:75:32:60:44:c9"`.
+
+A success response is of the form
+
+    {
+        "Interfaces": [{
+            "ID": int,
+            "Address": string,
+            "AddressIPv6": string,
+            "MacAddress": string
+        }, ...]
+    }
+
+with values in the `Interfaces` entries as above. For each entry, an `ID` and `MacAddress` and one of `Address` and `AddressIPv6` must be given. The `ID` is arbitrary but must differ among entries. It is used to identify, within the scope of the endpoint, an individual interface during a `Join` call.
+
+If the remote process was supplied entries in `Interfaces`, it must response with an empty `Interfaces` list.
+
```

Comment:
It would be good to be explicit and mention the fact that in this case if the driver returns an `Interfaces` list, it is considered as an error and it will result in a rollback operation.